### DISCUSSION TITLE
Potential issue regarding observation attributes

### DISCUSSION THREAD

**Discussion by MoBosse** [open-sdg/open-sdg#2096] [Q&A] [answered]
Hi all,

I recently upgraded one of my test platforms to 2.3.0. Mainly to utilise the functionality of the observation attributes.
For indicators without headline data, everything works fine. However, when I have data with empty columns, only the headline data is displayed, but not the breakdowns.

Year,sex,COMMENT_OBS,Value
2020,female,estimated,100
2020,male,estimated,90
...would work

Year,sex,COMMENT_OBS,Value
2020,,estimated,110
2020,female,estimated,100
2020,male,estimated,90
...would only show the data for total sex, not for male and female...

Year,sex,COMMENT_OBS,Value
2020,total,estimated,110
2020,female,estimated,100
2020,male,estimated,90
...would again work fine...

Could someone please check if this behaviour always occurs? I'm a bit unsure because I have a number of overwritten files so I don't know if the issue might be there.

**Comment by sarah-beasley** [comment]
Hi Moritz, sorry I'm not sure what the issue is here yet, as I've tried it on a feature branch and it seems to be working correctly for me.

e.g. i have added the observation attributes column with a few entries into data where it doesn't have sex or age breakdowns to replicate something similar to your situation - https://github.com/ONSdigital/sdg-data/blob/feature-2.3.0-beta3/data/indicator_1-2-2.csv

it has appeared as usual but with the observation attribute labelled as instructed on the site - http://uk-sdg-feature-branches.s3-website.eu-west-2.amazonaws.com/feature-2.3.0-beta3/1-2-2/

@brockfanning do you have any suggestions?

**Comment by MoBosse** [comment]
Hi Sarah, thanks for testing this. The fact that it works for you tells me that it's probably not a general bug, but a bug in my overwritten files. Could you please tell me where the site repo is that you used for testing? That would make it easier for me to identify the problem.

**Reply by sarah-beasley** [reply]
Sure, [here is the feature branch](https://github.com/ONSdigital/sdg-indicators/tree/feature-2.3.0-beta3) i was testing the functionality out in the site repo. That corresponds with[ this feature branch](http://uk-sdg-feature-branches.s3-website.eu-west-2.amazonaws.com/feature-2.3.0-beta3/1-2-2/) version of the site.

**Comment by MoBosse** [comment]
@sarah-beasley Could you do me another favor and test if your fetaure branch still works if you fill the COMMENT_OBS column for each row with a value such as "etimated". Like in the second example in my first post above.
After a bunch of testing, I found that this seems to trigger the problem (at least in my repos). 
See this commit for example: [https://github.com/SdgTestEnvironment/sdg-data/commit/458dcb24215fa7d992c498a520b6e72ab58264fe](https://github.com/SdgTestEnvironment/sdg-data/commit/458dcb24215fa7d992c498a520b6e72ab58264fe). With only one empty field in the COMMENT_OBS column it worked fine. After filling this field with an "estimated", only the headline data is shown.

**Reply by sarah-beasley** [reply]
Hi @MoBosse , I've tested and it seems so have worked ok for me... although I will say it didn't work in my browser when i first tested, but i put in another browser and saw the changes ok. Could be worth deleting the cookies/clearing the caching or trying another browser to test the change?

FYI - here is the file i changed to have an entry in each data line - https://github.com/ONSdigital/sdg-data/edit/feature-2.3.0-beta3/data/indicator_1-1-1.csv 
and here is the feature branch that i tested - http://uk-sdg-feature-branches.s3-website.eu-west-2.amazonaws.com/feature-2.3.0-beta3/1-1-1/

**Reply by MoBosse** [reply]
Thank you very much, but I'm afraid I expressed myself incorrectly. The error (if it occurs) is that only the main time series is displayed and not the disaggregations. However, the csv file you tested with does not contain any disaggregations. The error can therefore not appear here.

**Reply by MoBosse** [reply]
For example [this indicator](https://github.com/ONSdigital/sdg-data/blob/feature-2.3.0-beta3/data/indicator_1-2-2.csv) would only show the data for total age and sex if you give every line a value for the COMMENT_OBS column...

**Reply by sarah-beasley** [reply]
I see what you mean @MoBosse and having now tested for 1-2-2, am getting the same error, so perhaps it is a problem with Open SDG. I will raise with @brockfanning and see if he can fix!

Here is the data i changed for 1-2-2 - https://github.com/ONSdigital/sdg-data/blob/feature-2.3.0-beta3/data/indicator_1-2-2.csv

and here is the feature branch that i tested where the problem can now be seen (sex and age disaggregations no longer appear) - http://uk-sdg-feature-branches.s3-website.eu-west-2.amazonaws.com/feature-2.3.0-beta3/1-2-2/

**Reply by MoBosse** [reply]
Thank you. I was already doubting myself.

**Reply by sarah-beasley** [reply]
not at all, thanks for catching!

**Comment by brockfanning** [comment, marked as answer]
@sarah-beasley @MoBosse If you would like to preview a fix, PR #2102 should fix the bug.

**Reply by MoBosse** [reply]
Thank you @brockfanning .
Works!

**Comment by MoBosse** [comment]
@sarah-beasley I may have run into another issue similar to the [last one](https://github.com/open-sdg/open-sdg/issues/2097). Could you please check the following:
For an indicator with series toggle active, add comment_obs for one series but leave the column for the other series blank. For me, the chart shows the "no data available"-message when I switch to the second series. Can you confirm this?

**Reply by sarah-beasley** [reply]
Hi @MoBosse , i've given this a go and am not experiencing the same issue, but do let me know if you think i've not followed exactly. 

I have removed all the obs values for Series B of 1-3-1 and it seems to be working fine! See below:

Feature branch for testing - http://uk-sdg-feature-branches.s3-website.eu-west-2.amazonaws.com/feature-2.3.0-beta3/1-3-1/

Github FB - https://github.com/ONSdigital/sdg-data/blob/feature-2.3.0-beta3/data/indicator_1-3-1.csv

**Reply by MoBosse** [reply]
@sarah-beasley Thanks for testing but I think you missed one [here](https://github.com/ONSdigital/sdg-data/blob/feature-2.3.0-beta3/data/indicator_1-3-1.csv#L350)

**Reply by sarah-beasley** [reply]
@MoBosse ah, sorry! just waiting for it to load now i've removed :)

**Reply by MoBosse** [reply]
I see that it is still working. So it seems to be an issue somewhere in my code. Thanks anyway!

**Reply by sarah-beasley** [reply]
it's not finished loading yet so i will let you know when it's done and i've checked!

**Reply by sarah-beasley** [reply]
@MoBosse ah, it has finished loading now and it still seems to be fine for me

**Reply by sarah-beasley** [reply]
Hi @MoBosse is everything ok with this issue now?

**Reply by MoBosse** [reply]
Hi @sarah-beasley yes works fine for me now. Thanks!
